README.rst

```diff
@@ -186,13 +186,54 @@ writes the following data types to ``<output_dir>``:
 * CNA Continuous data
 * CNA Discrete data
 
-File structure, case lists and meta files will also be also added in the output folder.
+File structure, case lists and meta files will also be added to the output folder.
 See the  `cBioPortal file formats`_ documentation for further details.
 
 The output directory ``<output_dir>`` needs to be either empty or not yet existing.
 
 .. _`cBioPortal file formats`: https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats
 
+Source data assumptions and validation
+--------------------------------------
+
+General file characteristics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* ``Delimiter`` The source data should be provided as delimited text files. The delimiter can be `configured`_ per 
+  data file. If not configured, a tab-delimited file is assumed.
+* ``Comments`` Comment lines may be present, indicated by a ``#`` as the first character. These lines will be ignored.
+* ``Header`` The first non-comment line is assumed to be the header. It should be exactly one line.
+* ``Field number`` The number of fields (columns) is determined by the header. Every other line in the file 
+  should have this same number of fields (no blank lines).
+* ``Whitespace`` Leading or trailing whitespace is not trimmed. If present, it will persist in the final observation.
+* ``Encoding`` All files are assumed to be utf-8 encoded.
+
+CSR entities
+~~~~~~~~~~~~
+
+All characteristics and relationships of the CSR data model are defined in `csr/csr.py`_. Any field present in the
+source data that you would like to load to tranSMART, must be linked to a CSR field via the sources_config. Additional
+fields not present in the sources_config will be ignored.
+
+Regarding the source data, we can distinguish four types of validation:
+
+1. ``Value validation`` Independent validation of a single field value. This comprises type validation (e.g. string, integer or date), nullability (whether a field may be empty), and unique constraints.
+2. ``Record validation`` Validation across different fields from the same record within the same entity. This validation is relevant when the validity of a field value is dependent on the other fields of the same record (e.g. a biosource record with src_biosource_id = BS1, is invalid when biosource_id = BS1).
+3. ``Entity validation`` Concerns the integrity check of all records within a single entity (e.g. do all src_biosource_id values also have corresponding biosource_id records within the biosource entity).
+4. ``Across-entity validation`` Checks the validity of relationships between records of different entities.
+
+The data validation of the current pipeline is implemented for type 1 and to a limited extent for type 2 and 4.
+Hence, the source data is assumed to be coherent regarding its relationships within the same entity and across
+different entities. While erroneous relationships across entities, in respect of missing entity records, will be
```

Comment:
I'm not sure if this is entirely accurate, it seems that the following cases are detected:
- any entities referencing a non-existing patient;
- any biomaterials referencing a non-existing biosource.

Could be reformulated as 
> While some erroneous relationships across entities, in respect of missing entity records, will be detected (a biomaterial linked to a non-existing biosource or an entity linked to a non-existing patient), ...

Reply:
Ah indeed. The only exception being a non-existing diagnosis_id in the biosource entity?
Then I think we could change that to "most erroneous relationships"